src/index.ts

```diff
@@ -40,6 +52,52 @@ const createGenerator: CodegenGeneratorConstructor = (config, context) => {
 			return !!value.mediaType.mimeType.match('\\bjson$')
 		})
 
+		hbs.registerHelper('hasOperationSupportingMultipart', function(value: CodegenOperation | CodegenOperation[]): boolean {
+			if (isCodegenOperation(value)) {
+				return containsMultipartOperation([value])
+			} else {
+				return containsMultipartOperation(value)
+			}
+		})
+
+		hbs.registerHelper('isMultipartSchema', function(value: CodegenSchema): boolean {
+			return !!value.contentMediaType?.mimeType.match('^multipart/.*')
+		})
+
+		hbs.registerHelper('isMetadataSchema', function(value: CodegenSchema): boolean {
+			return value.purpose === CodegenSchemaPurpose.METADATA
+		})
+
+		hbs.registerHelper('getFileUploadProperties', function(properties: CodegenObjectLikeSchemas[]): FileUploadProperty[] {
+			const results: FileUploadProperty[] = []
+			for (const prop in properties) {
+				const property = properties[prop]
+				if (!isCodegenSchemaUsage(property)) {
+					continue
+				}
+
+				if (property.schema.purpose !== CodegenSchemaPurpose.METADATA) {
+					continue
+				}
+
+				if (isCodegenObjectSchema(property.schema) && property.schema.properties) {
```

Comment:
Do we need this test? I wonder if we instead check the array schema _first_ and then this is just the `else` block... we don't need to check it's an object schema here?

Reply:
Checked and it's fine.